internal/ingress/annotations/canary/main.go

```diff
@@ -30,10 +30,11 @@ type canary struct {
 
 // Config returns the configuration rules for setting up the Canary
 type Config struct {
-	Enabled bool
-	Weight  int
-	Header  string
-	Cookie  string
+	Enabled     bool
+	Weight      int
+	Header      string
+	HeaderValue string
```

Comment:
Adding `CookieValue` here would be good too for consistency or something more generic like mentioned by @ElvinEfendi in the email thread

Reply:
Should the change to be minimal to cover only the problem domain? The usual way to provide data is via headers and not cookies due security considerations (e.g. cookies are sent always and implicit). The @ElvinEfendi proposal need to be developed further to avoid complex logical constructions to avoid nginx performance penalties.

Reply:
Sure, I think it would be fine to handle the cookie case in a separate PR